index.bs

```diff
@@ -124,22 +131,61 @@ A conversion report is a [=struct=] with the following items:
 
 # Algorithms # {#algorithms}
 
-<h3 algorithm id="creating-conversion">Creating a conversion</h3>
-TODO
+<h3 algorithm id="creating-impression">Activating an impression</h3>
 
-<h3 algorithm id="creating-impression">Creating an impression</h3>
-TODO
+To <dfn>activate an impression</dfn> from an <{a}> tag,
+create a new [=impression=] struct whose items are:
+
+: [=impression/impression source=]
+:: The current top frame's [=origin=].
+: [=impression/impression data=]
+:: The result of applying [=parsing conversion data=] to the <{a/impressiondata}>
+    attribute.
+: [=impression/conversion destination=]
+:: The result of parsing <{a/conversiondestination}> as a site.
+: [=impression/reporting endpoint=]
+:: The result of [=origin/obtaining a site=] from the result of parsing
+    <{a/reportingorigin}> as an origin.
+: [=impression/expiry=]
+:: Be a timestamp <{a/impressionexpiry}> milliseconds in the future.
+: [=impression/impression time=]
+:: Be the current time.
+
+TODO: How to store the impression
+
+<h3 algorithm id="creating-a-conversion">Creating a conversion</h3>
+
+To <dfn>create a conversion</dfn> from a [=url=]:
+
+: [=conversion/conversion source=]
+:: The current top-level origin.
+: [=conversion/conversion data=]
+:: The result of applying [=parsing conversion data=] to the value associated with the
+    |conversion-data| field in the URL's query part.
+: [=conversion/conversion time=]
+:: Be the current time.
+
+TODO: Formalize how to parse the query similar to URLSearchParams.
 
 <h3 algorithm id="register-conversion">Register a conversion</h3>
-TODO
 
-<h3 algorithm id="parsing-the-metadata">Parsing the metadata</h3>
+To <dfn>register a conversion</dfn> from a fetch [=request=], run the following steps:
+
+1. If the requests [=request/current url's=] [=url/path=] is not `.well-known/register-conversion`,
+    return.
+1. If the request's [=request/redirect count=] is less than 1, return.
```

Comment:
It might be useful to include a `Note: ` here saying why you have this restriction.